.github/workflows/ci.yml

```diff
@@ -67,6 +76,19 @@ jobs:
     - name: Run pgx tests
       run: su postgres -c 'sh tools/build -pg${{ matrix.pgversion }} test-extension 2>&1'
 
+    - name: Build and install TimescaleDB
+      if: ${{ inputs.tsdb-commit != '' }}
+      run: |
+        apt-get install -y wget cmake gcc build-essential libkrb5-dev
+        git clone '${{ inputs.tsdb-repo }}'
+        cd timescaledb
+        git pull --all
```

Comment:
pull is fetch + merge and we discard the merge work on the next line.  But git fetch immediately after git clone is unnecessary as clone fetched everything already.  (Also --all is for all remotes but there's only one here).